### ISSUE TITLE
Feature request: Embed in iframe

### ISSUE THREAD

**Issue by ZaDarkSide** [gitpod-io/gitpod#419]
If you could embed in iframe like on https://stackblitz.com/docs#embedding that would be great.

**Comment by Alaboudi1** [comment]
I was actually going to use gitpod in a new website I planning to build, but it does not seem to support embedding in  iframe very well.

**Comment by trusktr** [comment]
Can you please not close issues with no activity? Otherwise valid and good issues get forgotten.

**Comment by akosyakov** [comment]
@trusktr Unfortunately it is hard for us to detect valid issues in all noise, please see for full explanation https://github.com/gitpod-io/gitpod/issues/3657#issue-844148077 Just go ahead and reopen the issue or create a new one.

**Comment by akosyakov** [comment]
I don't it is possible to implement in the full extend at least for VS Code. VS Code relies on browser APIs only available in the top window context and not from iframes.

**Comment by dlvhdr** [comment]
Hi @akosyakov,
I'm also trying to embed the openvscode-server in an iframe, but I'm getting "Forbidden." as the response.
When you said "In theory you can embed it", how would you manage to bypass this "Forbidden" error and why does it occur?

**Comment by akosyakov** [comment]
It sounds like you don't provide connection token as a part of iframe src.

**Comment by axonasif** [comment]
@vamche lets continue on your discord question

**Comment by trumbitta** [comment]
With the recent release of CodeSandbox Projects - their Gitpod-like experience - might this counter-attack be worth another look?

https://codesandbox.io/post/announcing-codesandbox-projects

**Comment by HadesArchitect** [comment]
Yep, for us embedding would be a valuable feature too, though I understand all the complications =\

**Comment by andreafalzetti** [comment]
> Yep, for us embedding would be a valuable feature too, though I understand all the complications =\

Could you elaborate more on how you would benefit from having this feature?

Considering that this task does not seem trivial at all, I think that sharing more details on why it's needed, could help understand how to prioritize this.

**Comment by HadesArchitect** [comment]
Ciao Andrea! 

Long story short, we have quite an edge case, so I doubt you'll really invest your time into that. It's a training platform over gitpod, my team works in the developer relations field to help engineers learn how to use Apache Cassandra, Apache Pulsar, etc. 

We would like to embed gitpod to our academy.datastax.com (open, free, oss-learning project) but looks like we will have to keep it "in the second tab" (one tab for academy, one tab for gitpod, integration via vscode extension pulling academy api)

<img width="1610" alt="image" src="https://user-images.githubusercontent.com/1742301/190445765-21923549-4e5c-4d4d-8024-278d84c88053.png">

Buona fortuna!

**Comment by vamche** [comment]
We have a similar use-case for tutorials, to embed the GitPod along with the learning material.

**Comment by loujaybee** [comment]
Updating title to remove "iframe" as there might be other solutions to the problem here, but it's not clear. If you're finding this issue and you need this for some reason, would be great to get the context behind the problem that you're solving.

**Comment by kamranayub** [comment]
Same context as @HadesArchitect, wanting to build an interactive course/tutorial experience. CodeSandbox is very good but unfortunately only their Browser Sandboxes are supported for embedding and the experience I'm helping on uses .NET, Python, Go, and Node.js -- so something that works beyond the browser would be excellent. GitPod was pretty easy to get started with but is missing some features for really making it ideal for learning scenarios.

For example, being able to pass some of the parameters CodeSandbox [allows for embedding](https://codesandbox.io/docs/learn/sandboxes/embedding) like `module` and `initialpath` would be awesome. Opening the VS editor on the right module with the right path in the browser would be an 80% improvement. For now I'm evaluating other alternatives, like creating separate branches, other embedding tech, etc.

**Comment by olaobajuabraham** [comment]
any news on this ?
